store/storetest/plugin_store.go

```diff
@@ -254,6 +252,18 @@ func doTestPluginCompareAndSet(t *testing.T, ss store.Store, s SqlSupplier, comp
 
 		actualKV, err := ss.Plugin().Get(kv.PluginId, kv.Key)
 		require.Nil(t, err)
+
+		// When tested with KVSetWithOptions, a strict comparison can fail because that
+		// function accepts a relative time and makes its own call to model.GetMillis(),
+		// leading to off-by-one issues. All these tests are written with 15+ second
+		// differences, so allow for an off-by-1000ms in either direction.
+		require.NotNil(t, actualKV)
+
+		expiryDelta := actualKV.ExpireAt - kv.ExpireAt
+		if expiryDelta > -1000 && expiryDelta < 1000 {
+			actualKV.ExpireAt = kv.ExpireAt
```

Comment:
ahh too bad you're testing the full struct, if it was just a single field, something like this would've worked as well:
```
require.InDelta(t, kv.ExpireAt, actualKV.ExpireAt, 1000)
actualKV.ExpireAt = kv.ExpireAt // ignore
require.Equal(t, kv, actualKV)
```

https://godoc.org/github.com/stretchr/testify/require#InDelta

Reply:
It would work, but currently fails on that one value if wrong, when the clue for the failure might be an entirely different key being returned. Probably a way around that too.